### ISSUE TITLE
Invalid "code against a concrete implementation" issue

### ISSUE THREAD

**Issue by edsonmedina** [scrutinizer-ci/php-analyzer#574]
I'm getting "Accessing name on the interface PhpParser\Node suggest that you code against a concrete implementation. How about adding an instanceof check?".

But I already have an "instanceof" on a separate function (which is invoked before the access)

see https://scrutinizer-ci.com/g/edsonmedina/php_testability/code-structure/master/class/edsonmedina%5Cphp_testability%5CNodeVisitors%5CClassVisitor?issueId=affe5ad7-5395-4b02-81d6-9b5d9e5fb54b

**Comment by donatj** [comment]
Getting this as well, although on a different package - see:

see https://scrutinizer-ci.com/g/donatj/PhpMySqlSchema/issues/master/files/src/Columns/AbstractColumn.php?orderField=path&order=asc&issueId=863835ad-1b56-4e31-81f8-6e192a8c0032

**Comment by edsonmedina** [comment]
Well, to be fair, there's no way any static analysis could ever catch that, so I'm closing this

**Comment by schmittjoh** [comment]
No, that's fine, let's keep this open. It's actually one the easier ones to implement :)
